src/condor_schedd.V6/schedd.cpp

```diff
@@ -17033,10 +17121,10 @@ Scheduler::RecycleShadow(int /*cmd*/, Stream *stream)
 	}
 #endif
 
-	if( !UserCheck2(NULL, cmd_user, match_user) ) {
+	if( !cmd_user || !UserCheck2(NULL, cmd_user, match_user) ) {
```

Comment:
This call to UserCheck2() looks wrong. It should grab the JobQueueJobAd of the job the shadow is done with for the first argument. And match_user can go away entirely.

Reply:
definitely wrong.  the old code passed a username to check against for the case where the first argument is null.  UserCheck2 should fail when the cmd_user and match_user are not the same.  We can do this check right after we fetch the two UserRecs instead of inside UserCheck2 now.